layout-server/src/test/java/org/finos/vuu/layoutserver/integration/LayoutIntegrationTest.java

```diff
@@ -0,0 +1,292 @@
+            .andExpect(jsonPath("$.metadata.group", is(layout.getMetadata().getGroup())))
+            .andExpect(jsonPath("$.metadata.screenshot", is(layout.getMetadata().getScreenshot())))
+            .andExpect(jsonPath("$.metadata.user", is(layout.getMetadata().getUser())));
+    }
+
+    @Test
+    void getLayout_validIDButLayoutDoesNotExist_returns404() throws Exception {
+        UUID layoutID = UUID.randomUUID();
+
+        mockMvc.perform(get("/layouts/{id}", layoutID)).andExpect(status().isNotFound());
+    }
+
+    @Test
+    void getLayout_invalidId_returns400() throws Exception {
+        String layoutID = "invalidUUID";
+
+        mockMvc.perform(get("/layouts/{id}", layoutID)).andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void getMetadata_metadataExists_returnsMetadata() throws Exception {
+        Layout layout = createDefaultLayoutInDatabase();
+
+        mockMvc.perform(get("/layouts/metadata"))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$[0].name", is(layout.getMetadata().getName())))
+            .andExpect(jsonPath("$[0].group", is(layout.getMetadata().getGroup())))
+            .andExpect(jsonPath("$[0].screenshot", is(layout.getMetadata().getScreenshot())))
+            .andExpect(jsonPath("$[0].user", is(layout.getMetadata().getUser())));
+    }
+
+    @Test
+    void getMetadata_metadataDoesNotExist_returnsEmptyList() throws Exception {
+        mockMvc.perform(get("/layouts/metadata"))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$").isEmpty());
+    }
+
+    @Test
+    void createLayout_validLayout_returnsLayoutCreatedWithIDAndCreatedDate() throws Exception {
+        LayoutRequestDTO layoutRequest = createValidCreateRequest();
+
+        mockMvc.perform(post("/layouts")
+                .content(objectMapper.writeValueAsString(layoutRequest))
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isCreated())
+            .andExpect(jsonPath("$.id").isNotEmpty())
+            .andExpect(jsonPath("$.created").isNotEmpty());
+    }
+
+
+    @Test
+    void createLayout_invalidLayout_returns400() throws Exception {
+        String invalidLayout = "invalidLayout";
+
+        mockMvc.perform(post("/layouts")
+                .content(invalidLayout)
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void createLayout_validLayoutButInvalidMetadata_returns400AndDoesNotCreateLayout()
+        throws Exception {
+        LayoutRequestDTO layoutRequest = createValidCreateRequest();
+        layoutRequest.setMetadata(null);
+
+        mockMvc.perform(post("/layouts")
+                .content(objectMapper.writeValueAsString(layoutRequest))
+                .contentType(MediaType.APPLICATION_JSON))
+            .andExpect(status().isBadRequest());
+
+        assertThat(layoutRepository.findAll()).isEmpty();
+    }
```

Comment:
> Let's make sure we test the transactional nature of this.
> https://github.com/ScottLogic/finos-vuu/pull/57#discussion_r1344149426

@pling-scottlogic Do you think this test sufficiently covers this?

Reply:
Yeah, that looks good to me. I think I would just do the same assertion on the metadata repository, for completeness.

Reply:
[bf7278e](https://github.com/ScottLogic/finos-vuu/pull/68/commits/bf7278e3202694a62712feabaf406c379640e73a)